docarray/index/backends/hnswlib.py

```diff
@@ -628,8 +627,11 @@ def accept_hashed_ids(id):
 
         # If hashed_ids is provided, k is the minimum of limit and the length of hashed_ids; else it is limit
         k = min(limit, len(hashed_ids)) if hashed_ids else limit
-
-        labels, distances = index.knn_query(queries, k=k, **extra_kwargs)
+        try:
+            labels, distances = index.knn_query(queries, k=k, **extra_kwargs)
+        except RuntimeError:
+            k = min(k, self.num_docs())
+            labels, distances = index.knn_query(queries, k=k, **extra_kwargs)
```

Comment:
we should just make sure that value of `k` is not exceeding number of vectors, no need for try catch

Reply:
I already modified the code so you can revert this change

Reply:
but this would involve having a call to num_docs very often which is what I am trying to avoid

Reply:
why does it matter so much it's cached

Reply:
I have seen it take 13 seconds for when the index is of the order of 1M documents. Just to solve a couple of very edge cases I do not think is worth